source/Cosmos.Core_Plugs/System/Collections/Generic/ComparerHelpersImpl.cs

```diff
@@ -1,117 +1,969 @@
+            {
+                return new NullableUInt64EqualityComparer();
+            }
+
+            if (aType == typeof(Char))
+            {
+                return new CharEqualityComparer();
+            }
+
+            if (aType == typeof(Char?))
+            {
+                return new NullableCharEqualityComparer();
+            }
+
+            if (aType == typeof(Int16))
+            {
+                return new Int16EqualityComparer();
+            }
+
+            if (aType == typeof(Int16?))
+            {
+                return new NullableInt16EqualityComparer();
+            }
+
+            if (aType == typeof(UInt16))
+            {
+                return new UInt16EqualityComparer();
+            }
+
+            if (aType == typeof(UInt16?))
+            {
+                return new NullableUInt16EqualityComparer();
+            }
+
+            if (aType == typeof(Guid))
+            {
+                return new GuidEqualityComparer();
+            }
+
+            if (aType == typeof(Guid?))
+            {
+                return new NullableGuidEqualityComparer();
+            }
+
+            if (aType.IsEnum)
+            {
+                switch (Type.GetTypeCode(Enum.GetUnderlyingType(aType)))
+                {
+                    case TypeCode.SByte:
+                        return new SByteEqualityComparer();
+                    case TypeCode.Byte:
+                        return new ByteEqualityComparer();
+                    case TypeCode.Int16:
+                        return new Int16EqualityComparer();
+                    case TypeCode.UInt16:
+                        return new UInt16EqualityComparer();
+                    case TypeCode.Int32:
+                        return new Int32EqualityComparer();
+                    case TypeCode.UInt32:
+                        return new UInt32EqualityComparer();
+                    case TypeCode.Int64:
+                        return new Int64EqualityComparer();
+                    case TypeCode.UInt64:
+                        return new UInt64EqualityComparer();
+                    default:
+                        return null;
+                }
+            }
+
+            //MS framework falls back to address compare so we'll do the same.
+            //mDebugger.Send($"No EqualityComparer for type {aType}");
+            //return null;
+            return new ObjectEqualityComparer();
+        }
+    }
+
+    #region "Comparer"
+
+    public class StringComparer : Comparer<string>
+    {
+        public override int Compare(string x, string y)
+        {
+            return String.Compare(x, y);
+        }
+    }
+
+    public class CharComparer : Comparer<Char>
+    {
+        public override int Compare(char x, char y)
+        {
+            return x.CompareTo(y);
+        }
+    }
+
+    public class NullableCharComparer : Comparer<Char?>
+    {
+        public override int Compare(Char? x, Char? y)

```

Comment:
I think the comparer is incorrectly handling nulls, as when only one of the two types is null, the other one is larger, rather than the two being equal.

Reply:
Added in master.